internal/bminventory/inventory.go

```diff
@@ -2181,6 +2167,54 @@ func (b *bareMetalInventory) UploadHostLogs(ctx context.Context, params installe
 	return installer.NewUploadHostLogsNoContent()
 }
 
+func (b *bareMetalInventory) DownloadHostLogs(ctx context.Context, params installer.DownloadHostLogsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Downloading logs from host %s in cluster %s", params.HostID, params.ClusterID)
+	hostObject, err := b.checkHostLogs(params.ClusterID.String(), params.HostID.String())
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+
+	fileName := b.getLogsFullName(params.ClusterID.String(), params.HostID.String())
+	exists, err := b.s3Client.DoesObjectExist(ctx, fileName)
```

Comment:
Why check if it exists first?  I would check that host.logs_collected_at is set, and then try to download.  If the download fails with error of NotFound, then return that error.

Reply:
still don't have this flag cause can't push due to problems with ignition image in subsystem. Waiting for @yevgeny-shnaidman that's why wanted to validate. Will delete